rfcs/get_signal.md

```diff
@@ -0,0 +1,88 @@
+# RFC 115: Add a new test method that returns AbortSignal
+
+## Summary
+
+Add a new test method called `get_signal` to provide AbortSignal that aborts
+when the test finishes.
+
+This method will throw if `AbortController` is not available.
```

Comment:
>Is that going to fail mysteriously?

I guess that'll be the case. Per MDN the version range with AbortController but not with addEventListener signal support:

* Firefox: 57-86
* Chrome: 66-90
* Safari: 12.1-15

Unfortunately there's no straightforward way to detect that, but this seemingly works:

```js
signalSupported = false;
this.addEventListener("", null, { get signal() { signalSupported = true; } });
signalSupported // becomes true
```

Reply:
Wouldn't it just be that the event handlers are invoked anyway, after the test has finished? Most of the time that is harmless, it's only if they poke at some shared state that it can affect subsequent tests.

Reply:
I agree that it'll be mostly harmless for event listeners. Anyway, there's a synchronous way to check the support, so I guess we are safe either way.

Reply:
So we can throw if there's no addEventListener support and that API is the main target of this proposal. Should we care about others e.g. Fetch and such?

Once we add get_signal then we probably won't be able to add more support checks for new APIs.

Reply:
cc @smaug----